app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,517 @@
+      startServerButton.setBackgroundColor(getResources().getColor(R.color.stopServer));
+    }
+  }
+
+  // This method checks if mobile data is enabled in user's device.
+  static boolean isMobileDataEnabled(Context context) {
+    ConnectivityManager cm =
+        (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    try {
+      Class cmClass = Class.forName(cm.getClass().getName());
+      Method method = cmClass.getDeclaredMethod("getMobileDataEnabled");
+      method.setAccessible(true);
+      return (Boolean) method.invoke(cm);
+    } catch (Exception e) {
+      Log.e(TAG, e.toString());
+    }
+    return false;
+  }
+
+  //This method sends the user to data usage summary settings activity
+  private void openDataUsageActivity() {
+    Intent intent = new Intent();
+    intent.setComponent(new ComponentName("com.android.settings",
+        "com.android.settings.Settings$DataUsageSummaryActivity"));
+    startActivity(intent);
+  }
+
+  @Override public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+      @NonNull int[] grantResults) {
+    switch (requestCode) {
+      case MY_PERMISSIONS_ACCESS_FINE_LOCATION: {
+        if (grantResults.length > 0
+            && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            toggleHotspot();
+          }
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    switch (requestCode) {
+      //Checking the result code for LocationSettings resolution
+      case LOCATION_SETTINGS_PERMISSION_RESULT:
+        final LocationSettingsStates states = LocationSettingsStates.fromIntent(data);
+        switch (resultCode) {
+          case Activity.RESULT_OK:
+            // All required changes were successfully made
+            Log.v(TAG, states.isLocationPresent() + "");
+            startService(ACTION_TURN_ON_AFTER_O);
+            break;
+          case Activity.RESULT_CANCELED:
+            // The user was asked to change settings, but chose not to
+            Log.v(TAG, "Canceled");
+            break;
+          default:
+            break;
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  @Override protected void onDestroy() {
+    super.onDestroy();
+    presenter.detachView();
+  }
+
+  private void setUpToolbar() {
+    Toolbar toolbar = findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    getSupportActionBar().setTitle(getString(R.string.menu_host_books));
+    getSupportActionBar().setHomeButtonEnabled(true);
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+    toolbar.setNavigationOnClickListener(v -> onBackPressed());
+  }
+
+  void setupLocationServices() {
+    LocationRequest mLocationRequest = new LocationRequest();
```

Comment:
Don't use Hungarian notations. We do have that in our contributors guidelines.

Reply:
Fixed. c2cb7feae30bfd149c3b6b486c4a3ddf8438c868